### PR TITLE
Epmlstrcmw-157: delete and get endpoints for project file

### PR DESCRIPTION
**Pull request by IaroslavTavchenkov**

### CODE REVIEW

File: controllers/src/test/scala/cromwell/pipeline/controller/ProjectFileControllerTest.scala [hunk 1]
```diff
@@ -77,5 +81,47 @@ class ProjectFileControllerTest extends AsyncWordSpec with Matchers with Scalate
...
+      "return OK response for valid request" taggedAs Controller in {
+        when(projectFileService.deleteFile(project, path, branchName, commitMessage)).
+          thenReturn(Future.successful(Right("Success")))
+        Delete(s"/files/try?$projectId&$projectFile") ~> projectFileController.route(accessToken) ~> check {
+          status shouldBe StatusCodes.OK
+        }
+      }
+    }
+
+    "get file" should {
```
Comment (IaroslavTavchenkov) [inline controllers/src/test/scala/cromwell/pipeline/controller/ProjectFileControllerTest.scala:108]: Println doesn't work here :(

Comment (GrigoriiBerezin) [reply controllers/src/test/scala/cromwell/pipeline/controller/ProjectFileControllerTest.scala:108]: https://github.com/IaroslavTavchenkov/cmwl_pipeline/pull/1

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala [hunk 2]
```diff
@@ -26,24 +30,62 @@ class ProjectFileController(wdlService: ProjectFileService)(implicit val executi
...
+                  case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
+                }
+            }
+          },
+          get {
+            parameter('projectId.as[String], 'path.as[String], 'version.as[String]) {
+              (projectId, path, version) =>
+                println(projectId)
+                println(path)
+                println(version)
```
Comment (GrigoriiBerezin) [inline controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala]: Must remove this

**Review by GrigoriiBerezin — CHANGES_REQUESTED** [review]
Great work, thanks. Just one moment to change.

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala (hunk 2 above)
Comment (IaroslavTavchenkov) [reply controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala]: Thank you!)

**Review by Dmitrii-Shevchenko — APPROVED** [review]
Good job!

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala [hunk 3]
```diff
@@ -28,24 +31,59 @@ class ProjectFileController(wdlService: ProjectFileService)(
...
+                    case (_, Left(response))         => StatusCodes.UnprocessableEntity.intValue -> response.message
+                  }
+                }) {
+                  case Success((status, message)) => complete(status, message)
+                  case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
+                }
+            }
+          },
+          get {
+            parameter('projectId.as[String], 'path.as[String], 'version.as[String]) {
```
Comment (iliaonishchenko) [inline controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala:54]: Didn't we agree that `version` query parameter should be optional? 
Also, this way of declaring parameters is deprecated in Akka-http now as far as I remember

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala [hunk 4]
```diff
@@ -28,24 +31,59 @@ class ProjectFileController(wdlService: ProjectFileService)(
...
+                  case Failure(e)       => complete(StatusCodes.InternalServerError, e.getMessage)
                 }
-              }) {
-                case Success((status, message)) => complete(status, message)
-                case Failure(e)                 => complete(StatusCodes.InternalServerError, e.getMessage)
-              }
+            }
+          },
+          delete {
+            parameter('projectId.as[String], 'path.as[String], 'branchName.as[String], 'commitMessage.as[String]) {
```
Comment (iliaonishchenko) [inline controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala:68]: Why're we specifying such things as `branchName` and `commitMessage` ?
User doesn't know that we have git under the hood, so we should not disclosure internals for anyone.

File: controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala [hunk 5]
```diff
@@ -28,24 +31,59 @@ class ProjectFileController(wdlService: ProjectFileService)(
...
+              (projectId, path, branchName, commitMessage) =>
+                onComplete(projectService.getProjectById(ProjectId(projectId)).flatMap {
+                  case Some(project) =>
+                    val future: Future[Either[VersioningException, String]] =
+                      wdlService.deleteFile(project, Paths.get(path), branchName, commitMessage)
```
Comment (iliaonishchenko) [inline controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala:73]: Why do we have this declaration?

File: controllers/src/test/scala/cromwell/pipeline/controller/ProjectFileControllerTest.scala [hunk 6]
```diff
@@ -1,23 +1,28 @@
...
 
+import java.net.URLEncoder
 import java.nio.file.Paths
 
 import akka.http.scaladsl.model.StatusCodes
 import akka.http.scaladsl.testkit.ScalatestRouteTest
-import cromwell.pipeline.datastorage.dao.repository.utils.{ TestProjectUtils, TestUserUtils }
-import cromwell.pipeline.datastorage.dto.auth.AccessTokenContent
+import cromwell.pipeline.datastorage.dao.repository.utils.{TestProjectUtils, TestUserUtils}
+//import cromwell.pipeline.datastorage.dto.ProjectBuildConfigurationRequest.ProjectUpdateFileRequest
```
Comment (iliaonishchenko) [inline controllers/src/test/scala/cromwell/pipeline/controller/ProjectFileControllerTest.scala:9]: Why do we need commented line?

File: repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala [hunk 7]
```diff
@@ -167,6 +168,17 @@ object FileContent {
...
+final case class ProjectGetFileRequest(project: Project, projectFile: ProjectFile, version: Option[PipelineVersion])
+
+object ProjectGetFileRequest {
+  implicit lazy val projectGetFileRequestFormat: OFormat[ProjectGetFileRequest] =
+    ((JsPath \ "project").format[Project] ~ (JsPath \ "projectFile").format[ProjectFile] ~ (JsPath \ "version")
+      .formatNullable[PipelineVersion])(
+      ProjectGetFileRequest.apply,
+      unlift(ProjectGetFileRequest.unapply)
+    )
+}
```
Comment (iliaonishchenko) [inline repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala:180]: Why do we need such complicated formatter? What is wrong with default?

**Review by iliaonishchenko — CHANGES_REQUESTED** [review]
Thank you for you work!
There're several things we need to speak about...

**Review by GrigoriiBerezin — CHANGES_REQUESTED** [review]
Great, thank you. Agree with Ilya's comments, will approve after him

**Pull request closed without merging**